tests/test_client_functional.py

```diff
@@ -2453,6 +2457,40 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url, error_message_url",
+    (
+        # yarl.URL.__new__ raises ValueError
+        ("http://:/", "http://:/"),
+        ("http://example.org:non_int_port/", "http://example.org:non_int_port/"),
+        # # yarl.URL.origin raises ValueError
+        ("http:/", "http:///"),
+        ("http:/example.com", "http:///example.com"),
+        ("http:///example.com", "http:///example.com"),
+        ("bluesky://d:i:d", "bluesky://d:i:d"),
+        ("call:+3801234567", r"call:\+3801234567"),
+        ("skype:handle", "skype:handle"),
+        ("slack://instance/room", "slack://instance/room"),
+        ("steam:code", "steam:code"),
+        ("twitter://handle", "twitter://handle"),
+    ),
+)
+async def test_invalid_redirect_url(
+    aiohttp_client: Any, invalid_redirect_url: Any, error_message_url: str
+) -> None:
+    headers = {hdrs.LOCATION: invalid_redirect_url}
+
+    async def handler_request(request):
+        return web.Response(status=301, headers=headers)
+
+    app = web.Application()
+    app.router.add_get("/redirect", handler_request)
+    client = await aiohttp_client(app)
+
+    with pytest.raises(InvalidRedirectURL, match=f"^{error_message_url} - [a-zA-Z]+"):
```

Comment:
Regexps are traditionally raw-strings:
```suggestion
    with pytest.raises(InvalidRedirectURL, match=fr"^{error_message_url} - [a-zA-Z]+"):
```